src/renderer/app.jsx

```diff
@@ -108,25 +99,44 @@ const ScratchDesktopHOC = function (WrappedComponent) {
         }
         handleTelemetryModalOptIn () {
             ipcRenderer.send('setTelemetryDidOptIn', true);
+            this.forceUpdate();
         }
         handleTelemetryModalOptOut () {
             ipcRenderer.send('setTelemetryDidOptIn', false);
+            this.forceUpdate();
         }
         handleUpdateProjectTitle (newTitle) {
             this.setState({projectTitle: newTitle});
         }
         render () {
-            const shouldShowTelemetryModal = (typeof ipcRenderer.sendSync('getTelemetryDidOptIn') !== 'boolean');
+            const currentTelemetryState = ipcRenderer.sendSync('getTelemetryDidOptIn');
```

Comment:
This is probably fine right now but could bite us in the future, since it's a blocking function call in `render`. We could avoid it by setting/fetching it in `this.state` instead of relying on fetching it here and rerendering with `forceUpdate`.